src/main/scala/firrtl/annotations/transforms/EliminateTargetPaths.scala

```diff
@@ -171,6 +257,47 @@ class EliminateTargetPaths extends Transform {
       newCircuit.copy(modules = modulesx)
     }
 
-    state.copy(circuit = newCircuitGC, renames = Some(renameMap), annotations = annotationsx)
+    // If instance targeted is in fact the only instance of a module, then it should not be renamed
+    // E.g. if Eliminate Target Paths on ~Top|Top/foo:Foo, but that is the only instance of Foo, then should return
+    //   ~Top|Top/foo:Foo, not ~Top|Top/foo:Foo___Top_foo
+    val renamedModuleMap = RenameMap(this)
+    val ct = CircuitTarget(newCircuitGC.main)
+    val newModule2Original = state.circuit.modules.map {
+      m => ct.module(m.name)
+    }.map {
+      t => (t, renameMap.get(ct.module(t.name)))
+    }.flatMap {
+      case (origMod, Some(Seq(newMod: IsModule))) =>
+        Some(Target.referringModule(newMod).module -> origMod.module)
+      case _ =>
+        None
+    }.toMap
```

Comment:
```suggestion
    val newModule2Original = state.circuit.modules.flatMap { m =>
      val t = ct.module(m.name)
      renameMap.get(ct.module(t.name))  // isn't t a ModuleTarget? how can we call .name on it?
               .collect { case Seq(newMod: IsModule) =>
                 Target.referringModule(newMod).module -> t.module
               }
   }.toMap
```
Or if you prefer the original, at least do `state.circuit.modules.view.map` so that we don't create all intermediate collections.